Oracle.md

```diff
@@ -0,0 +1,170 @@
+
+- The R value that will be used to sign the event outcome (possibly several).
+- The public key of the oracle.
+- The earliest date at which the event outcome will be released in Unix epoch format.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Three kinds of event outcomes are defined (note that a single even can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+For the sake of simplicity, outcomes should be represented as strings.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- end: the last possible outcome number
+- step: the increment
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+end: 100
+step: 1
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: the base in which the outcome value is decomposed.
+
+Prior to the event, the oracle releases `n + 1` R-values, where `n` is the expected number of digits that will make up the outcome value.
+The oracle releases an additional R-value with which it will sign the actual number of digits that make up the outcome value.
+This last value is meant for covering cases where the outcome value goes outside of the range that it was expected to stay in.
+The oracle should provide the R values as a list where the extra R value is placed first.
+
+#### Example: BTC/USD rate
+
+Alice and Bob wish to enter into a DLC at time `t` when BTC/USD ~= $10000 with a maturity time of `t+2`.
+The oracle at that time has a safety margin of one digit, and so provides 1 + 6 R-values:
+
+```
+base: 10
+RValues: [R0, R1, R2, R3, R4, R5]
+```
+
+Alice and Bob create a DLC with "relevant" outcomes between $10000 and $10999.
+Below $10000 Alice gets everything, and above $10999 Bob gets everything.
+They use the R0 value to cover the cases where BTC/USD < $10000 and BTC/USD > $10999.
+As they want to be safe, they cover cases up to a value of BTC/USD composed of 10 digits, which they can do with only a few extra CETs.
+
+They will thus create the following sets of CETs for the following cases:
+1. BTC/USD is in [0; 9], they use only `R0=1`, Alice gets everything
+2. BTC/USD is in [10; 99], they use only `R0=2`, Alice gets everything
+
+...
+
+4. BTC/USD is in [1000; 9999] they use only `R0=4`, Alice gets everything
+5. BTC/USD is in [10000; 10999] they use `R0=5, R1=1, R2=0, R3 in [0; 9], R4 in [0; 9], R5 in [0;9], the payouts are divided between Alice and Bob following some pre-determined distribution
```

Comment:
it might be useful to say here that if the payout is linear in the event value, then this line item can be expressed with 30 CETs, otherwise it requires 1000 CETs for full resolution.

Reply:
I'm not sure I follow you here can you explain a bit more in detail how you would cut it down to 30?

Reply:
If the total collateral is 0.666 BTC, then divide the collateral into three - 0.6, 0.06, 0.006.  Then distribute each denomination according to the matching R value (R3, R4, R5).

Reply:
Sorry I might be a bit tired, but I cannot see what you're trying to do here.

> if the payout is linear in the event value

For me that would mean that, assuming total collateral of 2 BTC if BTC/USD=10000 payout is Alice = 0.001 Bob = 1.999 then if BTC/USD=10001 payout is Alice = 0.002 Bob = 1.998... Do we have the same understanding?

> If the total collateral is 0.666 BTC, then divide the collateral into three - 0.6, 0.06, 0.006. Then distribute each denomination according to the matching R value (R3, R4, R5).

Don't really get this. For me you do need 1000 transactions since each payout is different (unless I misunderstood the linear part).

Reply:
It would require the funding transaction to have multiple outputs, one for every digit (digits 3,4,5 in this case).  Each output would have 10 CETs, one for each possible digit.  The sum could represent any outcome with those digits varying.

That said, the on-chain fees would be higher, so it might not be attractive overall.